onyo/commands/new.py

```diff
@@ -84,23 +113,26 @@ def create_filename(onyo_root, template):
     return filename
 
 
-def create_asset_file_cmd(directory, filename):
-    return "touch \"" + os.path.join(directory, filename) + "\""
-
+def sanitize_paths(directory, template, onyo_root):
+    """
+    Check and normalize the directory and template for the creation of a new
+    asset.
 
-def prepare_arguments(directory, template, onyo_root):
-    directory = os.path.join(onyo_root, directory)
+    Returns the absolute path of the directory and of the template on success.
+    """
+    directory = Path(onyo_root, directory).resolve()
     if not template:
         template = get_config_value('onyo.new.template', onyo_root)
 
-    template = os.path.join(get_git_root(onyo_root), os.path.join(".onyo/templates", template))
+    template = Path(get_git_root(onyo_root), ".onyo/templates", template).resolve()
 
     problem_str = ""
-    if not os.path.isfile(template):
-        problem_str = problem_str + "\nTemplate file " + os.path.join(get_git_root(onyo_root), os.path.join(".onyo/templates", template)) + " does not exist."
-    if not os.path.isdir(directory):
+    if not template.is_file():
+        problem_str = problem_str + "\nTemplate file " + template + " does not exist."
```

Comment:
I will. 

The idea of not erroring immediately is that onyo can print out a list of problems that can be solved at once before calling onyo again, instead of re-running the command multiple times and getting everytime a new error. That said, in some of the instances it is probably not necessary.

Reply:
There are ways to do that while not assigning everything to one generic variable.